src/presign/record.rs

```diff
@@ -96,14 +97,18 @@ impl PresignRecord {
         })
     }
 
-    pub(crate) fn sign(&self, d: sha2::Sha256) -> Result<(k256::Scalar, k256::Scalar)> {
+    /// Generate a signature share on the given `digest` with
+    /// the [`PresignRecord`].
+    pub fn sign(self, d: sha2::Sha256) -> Result<SignatureShare> {
```

Comment:
todo: you should pull over the `#[instrument...` and `info!` lines from the old sign function. It's for logging and since this is an important function, it's nice to know when it happened.

See the [Logging](https://github.com/boltlabs-inc/tss-ecdsa/blob/main/CONTRIBUTING.md#logging) section of the contributing file, especially the [guidance on which functions to instrument](https://github.com/boltlabs-inc/tss-ecdsa/blob/main/CONTRIBUTING.md#which-function-should-be-instrumented).